src/neural_de.jl

```diff
@@ -339,6 +339,138 @@ function (n::NeuralODEMM)(x, ps, st)
+
+Arguments:
+
+  - `model`: A `Flux.Chain` or `Lux.AbstractLuxLayer` neural network that defines the ODE
+    dynamics dh/dt = model(h). Should map hidden_size -> hidden_size.
+  - `cell`: A `Lux.AbstractRecurrentCell` (e.g., `LSTMCell`, `GRUCell`, `RNNCell`) that
+    processes inputs at each time step.
+  - `args`: Additional positional arguments passed to the ODE solver.
+  - `ordering`: The time series data batch ordering. Defaults to `BatchLastIndex()`.
+  - `return_sequence`: If `true`, returns outputs at all time steps. If `false`, returns
+    only the final output. Defaults to `false`.
+  - `sensealg`: The choice of differentiation algorithm used in the backpropagation.
+    Defaults to `InterpolatingAdjoint(; autojacvec = ZygoteVJP())`.
+  - `kwargs`: Additional keyword arguments splatted to the ODE solver.
+
+The forward pass takes a tuple `(x, ts)` where:
+  - `x`: Input array of shape `(input_dim, seq_len, batch)` with `BatchLastIndex()` ordering
+  - `ts`: Vector of time points corresponding to each element in the sequence
+
+Example:
+
+```julia
+using DiffEqFlux, Lux, Random
+
+rng = Random.default_rng()
+input_dim, hidden_dim, seq_len, batch_size = 2, 4, 10, 3
+
+# ODE dynamics model (hidden_dim -> hidden_dim)
+ode_model = Chain(Dense(hidden_dim => hidden_dim, tanh))
+
+# RNN cell (processes input_dim -> hidden_dim)
+cell = LSTMCell(input_dim => hidden_dim)
+
+# Create ODE-RNN layer
+odernn = ODERNN(ode_model, cell; return_sequence=true)
+
+# Setup
+ps, st = Lux.setup(rng, odernn)
+
+# Input data and time points
+x = randn(Float32, input_dim, seq_len, batch_size)
+ts = collect(Float32, range(0, 1, length=seq_len))
+
+# Forward pass
+outputs, st = odernn((x, ts), ps, st)
+```
+
+References:
+
+[1] Rubanova, Yulia, Ricky TQ Chen, and David Duvenaud. "Latent ordinary differential
+    equations for irregularly-sampled time series." Advances in neural information
+    processing systems 32 (2019).
+
+[2] Lechner, Mathias, and Ramin Hasani. "Learning long-term dependencies in irregularly-sampled
+    time series." arXiv preprint arXiv:2006.04418 (2020).
+"""
+@concrete struct ODERNN <: AbstractLuxContainerLayer{(:model, :cell)}
+    model <: AbstractLuxLayer
+    cell <: AbstractLuxLayer
+    return_sequence
+    ordering
+    args
+    kwargs
+end
+
+function ODERNN(model, cell, args...;
+        ordering = Lux.BatchLastIndex(),
+        return_sequence::Bool = false,
+        kwargs...)
+    !(model isa AbstractLuxLayer) && (model = FromFluxAdaptor()(model))
+    !(cell isa AbstractLuxLayer) && (cell = FromFluxAdaptor()(cell))
+    return ODERNN(model, cell, static(return_sequence), ordering, args, kwargs)
+end
+
+function (odernn::ODERNN)((x, ts)::Tuple{<:AbstractArray, <:AbstractVector}, ps, st)
+    xs = Lux.safe_eachslice(x, odernn.ordering)
+
+    # Initialize with first input to get initial hidden state
+    x0 = first(xs)
+    (_, init_carry), st_cell = odernn.cell(x0, ps.cell, st.cell)
```

Comment:
you can get the initial hidden state with `Lux.init_recurrent_state` without doing the first pass here